docs/Lexicon.rst

```diff
@@ -235,6 +269,10 @@ source code, tests, and commit messages. See also the `LLVM lexicon`_.
     IR generation.
     See `mandatory passes <mandatory passes / mandatory optimizations>`.
 
+  reabstraction
+    An implicit representation change that occurs when a value is used with
+    a different `abstraction pattern` from its current representation.
```

Comment:
Ah, do we not consider things reabstractions unless they change a representation like this? (e.g. Optional to non-Optional in covariant overrides)